File: 1706-min-cost-to-connect-all-points/min-cost-to-connect-all-points.cpp
class Solution {
public:
    int minCostConnectPoints(vector<vector<int>>& points) {
        int n=points.size();
        vector<vector<pair<int,int>>> adj(n);
        for(int i=0;i<n;i++)
        {
            for(int j=i+1;j<n;j++)
            {
                adj[i].push_back({j,abs(points[i][0]-points[j][0])+abs(points[i][1]-points[j][1])});
                adj[j].push_back({i,abs(points[i][0]-points[j][0])+abs(points[i][1]-points[j][1])});
            }
        }
        priority_queue<pair<int,int>,vector<pair<int,int>>,greater<pair<int,int>>> q;
        int sum=0;
        vector<int> cost(n,0);
        cost[0]=0;
        q.push({0,0});
        while(!q.empty())
        {
            int cst=q.top().first;
            int node=q.top().second;
            q.pop();
            if(cost[node]) continue;

            cost[node]=1;
            sum+=cst;

            for(auto& i : adj[node])
            {
                if(cost[i.first]==0)
                {
                    q.push({i.second,i.first});
                }
            }
        }
        return sum;
    }
  
};

File: 1706-min-cost-to-connect-all-points/min-cost-to-connect-all-points.java
class Solution {
    public int minCostConnectPoints(int[][] points) {
       int sum=0;
        PriorityQueue<int[]> pq = new PriorityQueue<>((a, b) -> a[0] - b[0]);
       int n=points.length;
       pq.offer(new int[]{0,0,-1});
       int[] visited = new int[n];
       Arrays.fill(visited,-1);
       while(!pq.isEmpty())
       {
         int[] top = pq.poll();
         int a=top[0];
         int b=top[1];
         int c=top[2];
         if(visited[b]==1) continue ;
         sum+=a;
         visited[b]=1;
         for(int j=0;j<n;j++)
         {
            if(b!=j && visited[j]==-1)
            {
                int dist = Math.abs(points[b][0]-points[j][0])+ Math.abs(points[b][1]-points[j][1]);
                pq.offer(new int[]{dist,j,b});
            }
         }

       }
       return sum;
    }
}